### DISCUSSION TITLE
Fault current mismatch

### DISCUSSION THREAD

**Discussion by dh-jacosta** [dss-extensions/dss-extensions#42] [Q&A] [answered]
Hello,

I am running a simple fault study for a small LV network. The purpose of my study is to compare it with another model for validation using OpenDSS. When I use the COM interface on Windows I get a set of results but using OpenDSSDirect I get differnt results. Does anybody have an idea why? This is what I am running:

dss(['clear',
'New Circuit.dss_LV basekv=11 pu=1.02671 angle=30 basefreq=50 phases=3 model=ideal',
'new transformer.LVSS windings=2 buses=(sourcebus, T1gnd.1.2.3.4) conns=(delta, wye) kvs=(11,0.415) kvas=(500, 500) %loadloss=0 xhl=0.000001',
'New linecode.default units=km nphases=4 basefreq=50 Rmatrix=[0.0001 |0 0.0001| 0 0 0.0001| 0 0 0 0.0001] Xmatrix=[0.0001 |0 0.0001| 0 0 0.0001| 0 0 0 0.0001] Cmatrix=[0 |0 0| 0 0 0| 0 0 0 0] Neutral=4 Kron=n',
'New linecode.trafo units=km nphases=4 basefreq=50 Rmatrix=[0.00509 |0 0.00509| 0 0 0.00509| 0 0 0 1e-8] Xmatrix=[0.0171 |0 0.0171| 0 0 0.0171| 0 0 0 0] Cmatrix=[0 |0 0| 0 0 0| 0 0 0 0] Neutral=4 Kron=n',
'New linecode.ideal units=km nphases=4 basefreq=50 Rmatrix=[1e-8 |0 1e-8| 0 0 1e-8| 0 0 0 1e-8] Xmatrix=[1e-8 |0 1e-8| 0 0 1e-8| 0 0 0 0] Cmatrix=[0 |0 0| 0 0 0| 0 0 0 0] Neutral=4 Kron=n',
'New linecode.0.5_CU units=km nphases=4 basefreq=50 Rmatrix=[0.073 |0 0.073| 0 0 0.073| 0 0 0 0.186] Xmatrix=[0.064 |0 0.064| 0 0 0.064| 0 0 0 0.064] Cmatrix=[0 |0 0| 0 0 0| 0 0 0 0] Neutral=4 Kron=n',
'New linecode.0.25_CU units=km nphases=4 basefreq=50 Rmatrix=[0.153 |0 0.153| 0 0 0.153| 0 0 0 0.153] Xmatrix=[0.067 |0 0.067| 0 0 0.067| 0 0 0 0.067] Cmatrix=[0 |0 0| 0 0 0| 0 0 0 0] Neutral=4 Kron=n',
'New linecode.0.15_AL units=km nphases=4 basefreq=50 Rmatrix=[0.394 |0 0.394| 0 0 0.394| 0 0 0 0.394] Xmatrix=[0.07 |0 0.07| 0 0 0.07| 0 0 0 0.07] Cmatrix=[0 |0 0| 0 0 0| 0 0 0 0] Neutral=4 Kron=n',
'New linecode.0.1_AL units=km nphases=4 basefreq=50 Rmatrix=[0.596 |0 0.596| 0 0 0.596| 0 0 0 0.569] Xmatrix=[0.073 |0 0.073| 0 0 0.073| 0 0 0 0.073] Cmatrix=[0 |0 0| 0 0 0| 0 0 0 0] Neutral=4 Kron=n',
'New Line.DistTransformer__1 Phases=4 Bus1=T1gnd.1.2.3.4 Bus2=1.1.2.3.4 Length=1 units=km Linecode= trafo',
'new line.1 phases=4 bus1=1.1.2.3.4 bus2=2.1.2.3.4 linecode=0.5_CU length=0.1 units=km',
'new line.2 phases=4 bus1=2.1.2.3.4 bus2=3.1.2.3.4 linecode=0.25_CU length=34.8 units=m',
'new line.3 phases=4 bus1=3.1.2.3.4 bus2=4.1.2.3.4 linecode=0.25_CU length=13.2 units=m',
'new line.4 phases=4 bus1=4.1.2.3.4 bus2=5.1.2.3.4 linecode=0.25_CU length=29.1 units=m',
'new line.5 phases=4 bus1=5.1.2.3.4 bus2=6.1.2.3.4 linecode=0.25_CU length=11.1 units=m',
'new line.7 phases=4 bus1=5.1.2.3.4 bus2=8.1.2.3.4 linecode=0.15_AL length=57.2 units=m',
'new line.6 phases=4 bus1=6.1.2.3.4 bus2=7.1.2.3.4 linecode=0.25_CU length=6.7 units=m',
'new line.8 phases=4 bus1=8.1.2.3.4 bus2=9.1.2.3.4 linecode=0.15_AL length=20.7 units=m',
'new line.9 phases=4 bus1=8.1.2.3.4 bus2=10.1.2.3.4 linecode=0.1_AL length=36.4 units=m',
'new line.10 phases=4 bus1=10.1.2.3.4 bus2=11.1.2.3.4 linecode=0.1_AL length=39.6 units=m',
'new fault.F1 Bus1=11.3 Bus2=11.4 phases=1 R=1e-8',
'set mode=snapshot',
'set voltagebases=(11 0.415)',
'calcvoltagebases',
'Solve'])

dss.Circuit.SetActiveElement(f"Fault.F1")
currents = dss.CktElement.Currents()

**Comment by PMeira** [comment]
I gave it a try and got nearly the same results on all platforms. With DSS-Extensions, latest versions:

- Windows x64: `[-196.69155883789062, 1206.4195575714111, 196.69155883789062, -1206.4195575714111]`
- Linux x64: `[-196.69155979156494, 1206.4195575714111, 196.69155979156494, -1206.4195575714111]`
- macOS ARM64/M1: `[-196.6915578842163, 1206.4195575714111, 196.6915578842163, -1206.4195575714111]`

With OpenDSS "Version 9.8.0.1 (64-bit build)" via COM: `[-196.6915578842163, 1206.4195556640625, 196.6915578842163, -1206.4195556640625]`

>  When I use the COM interface on Windows I get a set of results but using OpenDSSDirect I get differnt results. Does anybody have an idea why?

Depends what you mean by different. Can you confirm you are seeing the same results as above? If not, it's probably worth checking which versions of the packages you're using. 

I have some more comments but I'll post them after we confirm the basics are OK on your side.

**Reply by dh-jacosta** [reply]
I can confirm I get identical results as you:

- ubuntu x64: [-196.69155979156494, 1206.4195575714111, 196.69155979156494, -1206.4195575714111]
- macOS ARM64/M2: [-196.6915578842163, 1206.4195575714111, 196.6915578842163, -1206.4195575714111]

However, when using OpenDSS directly, I am getting the following:

ELEMENT = "Fault.F1"
11            3           1246.7 /_  102.5 =    -269.83 +j    1217.1
------------
11            4           1246.7 /_  -77.5 =     269.83 +j   -1217.1

I am not sure if I am taking the wrong results.

**Comment by PMeira** [comment, marked as answer]
Add `set defaultbasefreq=50` before creating the circuit. I think that's the main issue.

Without that, the currents also match with the default settings (60Hz). Comparing the GUI results using `visualize currents fault.f1`:

![`visualize currents fault.f1`](https://github.com/dss-extensions/dss-extensions/assets/10246101/a7c2d6a8-4617-4d54-9471-402864bdaf14)

Using 50Hz:

![image](https://github.com/dss-extensions/dss-extensions/assets/10246101/7926b1dd-22ee-463c-9358-551bd87939c1)

> The purpose of my study is to compare it with another model for validation using OpenDSS.

I understand you're comparing to something else, but remember that the basic data is not realistic, e.g. lines with diagonal matrices with no capacitance.

Besides that, just a more general comment: try to avoid using tiny impedances too much since the system can get ill conditioned. That would also result in numeric differences between the current OpenDSS GUI and the other alternatives (including the official COM DLL).

**Reply by dh-jacosta** [reply]
Thank you very much Paulo, I was very puzzled and I was sure I was doing something wrong. I appreciate it, this validation is very much an ideal modelling, I am not planning to keep these unrealistic values going forward.

Thanks Again!
